### PR TITLE
[SYCLomatic] Update query-api-mapping feature according to doc repo.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

**Review by ziranzha — APPROVED** [review]

File: clang/lib/DPCT/ApiMapping/runtime-api.inc [hunk 1]
```diff
@@ -0,0 +1,1120 @@
...
+"    #include <dpct/dpct.hpp>\n"
+"    int main() {\n"
+"      dpct::get_current_device().queues_wait_and_throw();\n"
+"    }\n"
+)
+REGISTER_ENTRY(
+"cudaStreamAddCallback",
+"CUDA API:\n"
+"    cudaError_t cudaStreamAddCallback(cudaStream_t stream, cudaStreamCallback_t callback, void* userData, unsigned int flags)\n"
+"Migrates to SYCL API:\n"
```
Comment (zhimingwang36) [inline clang/lib/DPCT/ApiMapping/runtime-api.inc:200]:
```suggestion
"Migrates to C++ Standard Library API:\n"
```

File: clang/lib/DPCT/ApiMapping/runtime-api.inc [hunk 2]
```diff
@@ -0,0 +1,1120 @@
...
+"      dpct::queue_ptr s;\n"
+"      s = dev_ct1.create_queue();\n"
+"      dev_ct1.destroy_queue(s);\n"
+"    }\n"
+)
+REGISTER_ENTRY(
+"cudaStreamGetFlags",
+"CUDA API:\n"
+"    cudaError_t cudaStreamGetFlags(cudaStream_t hStream, unsigned int* flags)\n"
+"Migrates to SYCL API:\n"
```
Comment (zhimingwang36) [inline clang/lib/DPCT/ApiMapping/runtime-api.inc:311]:
```suggestion
"Migrates to an expression statement which set the output parameter “flags” to 0\n"
```

File: clang/lib/DPCT/ApiMapping/runtime-api.inc [hunk 3]
```diff
@@ -0,0 +1,1120 @@
...
+"      The output parameter(s) are set to 0.\n"
+"      */\n"
+"      *(&flags) = 0;\n"
+"    }\n"
+)
+REGISTER_ENTRY(
+"cudaStreamGetPriority",
+"CUDA API:\n"
+"    cudaError_t cudaStreamGetPriority(cudaStream_t hStream, int* priority)\n"
+"Migrates to SYCL API:\n"
```
Comment (zhimingwang36) [inline clang/lib/DPCT/ApiMapping/runtime-api.inc:338]:
```suggestion
"Migrates to 0:\n"
```

File: clang/lib/DPCT/ApiMapping/runtime-api.inc [hunk 4]
```diff
@@ -0,0 +1,1120 @@
...
+"      dpct::event_ptr e;\n"
+"      e = new sycl::event();\n"
+"      (int)e->get_info<sycl::info::event::command_execution_status>();\n"
+"    }\n"
+)
+REGISTER_ENTRY(
+"cudaEventRecord",
+"CUDA API:\n"
+"    cudaError_t cudaEventRecord(cudaEvent_t event, cudaStream_t stream = 0)\n"
+"Migrates to SYCL API:\n"
```
Comment (zhimingwang36) [inline clang/lib/DPCT/ApiMapping/runtime-api.inc:494]:
```suggestion
"Migrates to C++ Standard Library API:\n"
```

File: clang/lib/DPCT/ApiMapping/runtime-api.inc [hunk 5]
```diff
@@ -0,0 +1,1120 @@
...
+"      s = dev_ct1.create_queue();\n"
+"      dev_ct1.destroy_queue(s);\n"
+"    }\n"
+)
+REGISTER_ENTRY(
+"cudaStreamGetFlags",
+"CUDA API:\n"
+"    cudaError_t cudaStreamGetFlags(cudaStream_t hStream, unsigned int* flags)\n"
+"Migrates to SYCL API:\n"
+"    =\n"
```
Comment (zhimingwang36) [inline clang/lib/DPCT/ApiMapping/runtime-api.inc:312]:
```suggestion
"    \n"
```

File: clang/lib/DPCT/ApiMapping/runtime-api.inc [hunk 6]
```diff
@@ -0,0 +1,1120 @@
...
+"      e = new sycl::event();\n"
+"      dpct::destroy_event(e);\n"
+"    }\n"
+)
+REGISTER_ENTRY(
+"cudaEventQuery",
+"CUDA API:\n"
+"    cudaError_t cudaEventQuery(cudaEvent_t event)\n"
+"Migrates to SYCL API:\n"
+"    detail::is_event_info_desc<Param>::return_type sycl::event::get_info()\n"
```
Comment (zhimingwang36) [inline clang/lib/DPCT/ApiMapping/runtime-api.inc:474]:
```suggestion
"    sycl::detail::is_event_info_desc<Param>::return_type sycl::event::get_info()\n"
```

File: clang/lib/DPCT/ApiMapping/runtime-api.inc [hunk 7]
```diff
@@ -0,0 +1,1120 @@
...
+"      p = (float *)sycl::malloc_device(sizeof(p), q_ct1);\n"
+"      sycl::free(p, q_ct1);\n"
+"    }\n"
+)
+REGISTER_ENTRY(
+"cudaGetSymbolAddress",
+"CUDA API:\n"
+"    cudaError_t cudaGetSymbolAddress(void** devPtr, const void* symbol)\n"
+"Migrates to helper function:\n"
+"    value_t *dpct::detail::device_memory::get_ptr()\n"
```
Comment (zhimingwang36) [inline clang/lib/DPCT/ApiMapping/runtime-api.inc:731]: for detail:: namespace,  it should be not used in migration code directly,  so here need update to use the publich availalbe type in dpct:: namespace.

**Pull request closed without merging**